src/bin/ch-remote.rs

```diff
@@ -652,334 +895,239 @@ fn create_data(path: &str) -> Result<String, Error> {
-    /// memory zone identifier
-    id: String,
-
-    #[argh(option, long = "size")]
-    /// new memory size in bytes (supports K/M/G suffix)"
-    size: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "snapshot")]
-/// Create a snapshot from VM
-struct SnapshotSubcommand {
-    #[argh(positional)]
-    /// destination_url
-    snapshot_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "restore")]
-/// Restore VM from a snapshot
-struct RestoreSubcommand {
-    #[argh(positional)]
-    /// restore config
-    restore_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "coredump")]
-/// Create a coredump from VM
-struct CoredumpSubcommand {
-    #[argh(positional)]
-    /// coredump config
-    coredump_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "send-migration")]
-/// Initiate a VM migration
-struct SendMigrationSubcommand {
-    #[argh(switch, long = "local")]
-    /// local migration
-    send_migration_local: bool,
-
-    #[argh(positional)]
-    /// destination_url
-    send_migration_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "receive-migration")]
-/// Receive a VM migration
-struct ReceiveMigrationSubcommand {
-    #[argh(positional)]
-    /// receiver url
-    receive_migration_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "create")]
-/// Create a VM from a JSON configuration
-struct CreateSubcommand {
-    #[argh(positional, default = "String::from(\"-\")")]
-    /// vm config
-    vm_config: String,
-}
-
-#[derive(FromArgs, PartialEq, Debug)]
-#[argh(subcommand, name = "version")]
-/// Print version information
-struct VersionSubcommand {}
-
 fn main() {
-    let toplevel: TopLevel = argh::from_env();
+    let app = Command::new("ch-remote")
+        .author(env!("CARGO_PKG_AUTHORS"))
+        .subcommand_required(true)
+        .about("Remotely control a cloud-hypervisor VMM.")
+        .args([
+            Arg::new("api-socket")
+                .long("api-socket")
+                .help("HTTP API socket path (UNIX domain socket).")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-service-name")
+                .long("dbus-service-name")
+                .help("Well known name of the dbus service")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-object-path")
+                .long("dbus-object-path")
+                .help("Object path which the interface is being served at")
+                .num_args(1),
+            #[cfg(feature = "dbus_api")]
+            Arg::new("dbus-system-bus")
+                .long("dbus-system-bus")
+                .action(clap::ArgAction::SetFalse)
+                .help("Use the system bus instead of a session bus")
+                .num_args(1),
```

Comment:
Hey @rveerama1, sorry for the late reply.

The issues related to the D-Bus API tests seem to be stemming from here. I've not used `clap` before, but `clap::ArgAction::SetFalse` sets this flag to `false` when it's present--it should be the opposite. Because of this, `ch-remote` is trying to connect to the system bus when running integration tests.

Also, since this is a binary flag, there's no need to call `num_args(1)` on it. `ch-remote` panics at runtime because of this if you try running `./ch-remote --help`

Reply:
@brkp thanks for the reply. Modified ArgAction and num_args.
When I run tests it fails.

```
==== Start cloud-hypervisor command-line ====

"target/x86_64-unknown-linux-gnu/release/cloud-hypervisor" "--dbus-service-name=org.cloudhypervisor.chK9LI99" "--dbus-object-path=/org/cloudhypervisor/chK9LI99" "-v"

==== End cloud-hypervisor command-line ====


Error running ch-remote command: "target/x86_64-unknown-linux-gnu/release/ch-remote" "--dbus-service-name=org.cloudhypervisor.chK9LI99" "--dbus-object-path=/org/cloudhypervisor/chK9LI99" "ping"
stderr: Error running command: Error D-Bus proxy: org.freedesktop.DBus.Error.ServiceUnknown: The name org.cloudhypervisor.chK9LI99 was not provided by any .service files

thread 'dbus_api::test_api_dbus_create_boot' panicked at 'assertion failed: target_api.remote_command(\"ping\", None)', tests/integration.rs:157:5

```